services/repository-managers/src/test/java/org/sagebionetworks/evaluation/manager/ParticipantManagerTest.java

```
@@ -0,0 +1,132 @@
+    	when(mockCompetitionManager.getEvaluation(eq(EVAL_ID))).thenReturn(eval);
+    	when(mockCompetitionManager.isEvalAdmin(eq(OWNER_ID), eq(EVAL_ID))).thenReturn(true);
+        
+        // Participant Manager
+    	participantManager = new ParticipantManagerImpl(mockParticipantDAO, mockUserManager, mockCompetitionManager);
+    }
+	
+    @Test
+    public void testCRDAsAdmin() throws NotFoundException {
+    	participantManager.addParticipantAsAdmin(ownerInfo, EVAL_ID, USER_ID);
+    	participantManager.getParticipant(USER_ID, EVAL_ID);
+    	participantManager.removeParticipant(ownerInfo, EVAL_ID, USER_ID);
+    	verify(mockParticipantDAO).create(eq(part));
+    	verify(mockParticipantDAO, times(2)).get(eq(USER_ID), eq(EVAL_ID));
+    	verify(mockParticipantDAO).delete(eq(USER_ID), eq(EVAL_ID));
+    }
+    
+    @Test
+    public void testCRDAsAdmin_NotOpen() throws NotFoundException {
+    	// admin should be able to add participants even if Evaluation is closed
+    	eval.setStatus(EvaluationStatus.CLOSED);
+    	participantManager.addParticipantAsAdmin(ownerInfo, EVAL_ID, USER_ID);
+    	participantManager.getParticipant(USER_ID, EVAL_ID);
+    	participantManager.removeParticipant(ownerInfo, EVAL_ID, USER_ID);
+    	verify(mockParticipantDAO).create(eq(part));
+    	verify(mockParticipantDAO, times(2)).get(eq(USER_ID), eq(EVAL_ID));
+    	verify(mockParticipantDAO).delete(eq(USER_ID), eq(EVAL_ID));
+    }
+    
+    @Test
+    public void testCRDAsUser() throws DatastoreException, NotFoundException {
+    	participantManager.addParticipant(userInfo, EVAL_ID);
+    	participantManager.getParticipant(USER_ID, EVAL_ID);
+    	participantManager.removeParticipant(userInfo, EVAL_ID, USER_ID);
+    	verify(mockParticipantDAO).create(eq(part));
+    	verify(mockParticipantDAO, times(2)).get(eq(USER_ID), eq(EVAL_ID));
+    	verify(mockParticipantDAO).delete(eq(USER_ID), eq(EVAL_ID));
+    }
+    
+    @Test(expected=IllegalStateException.class)
```

'addParticipant' should throw an unauthorized exception with a message that explains to the user that they are trying to join an Evaluation which is not yet open.